src/noit_check_rest.c

```diff
@@ -1239,6 +1244,32 @@ rest_show_check_updates(mtev_http_rest_closure_t *restc,
   return 0;
 }
 
+void
+rest_check_get_attrs(xmlNodePtr attr, char **target, char **name, char **module) {
+  xmlNodePtr a = NULL;
+  for(a = attr->children; a; a = a->next) {
+    if(!strcmp((char *)a->name, "target"))
+      *target = (char *)xmlNodeGetContent(a);
+    if(!strcmp((char *)a->name, "name"))
+      *name = (char *)xmlNodeGetContent(a);
+    if(!strcmp((char *)a->name, "module"))
+      *module = (char *)xmlNodeGetContent(a);
```

Comment:
These should be elseif's, since it's impossible for them to match if the first one did.  

Also, if debug mode is one, should we consider print that there were unknown arttributes as a final else and nothing matched?

Reply:
We're going to encountered multiple things that may not match - this is a case where we only want these three.

I added safety to make sure we're setting null for these if nothing matches and added a print for if we couldn't find any of them in the debug log.

Reply:
Removed the debug prints - "not found" is actually a reasonable/normal outcome for this and we handle those cases.